pkg/applicationserver/io/packages/loradms/v1/api/objects/objects.go

```diff
@@ -52,9 +52,107 @@ func (d DeviceUplinkResponses) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
+type PositionSolutionType uint8
+
+const (
+	// GNSSPositionSolutionType is GNSS position solution type.
+	GNSSPositionSolutionType PositionSolutionType = iota
```

Comment:
Should we string type this? I don't see the numerics being used here

Reply:
The advantage in doing this is that the 0-value has the same semantics as the DMS response. By default a solution type of `gnss` is not going to be reported by the API - it's implicit that a location solution without an `algorithm_type` type is of type `gnss`. See the following example of a `gnss` location solution:

```json
{
	"accuracy": 8.2,
	"capture_time_gps": 1291395453.925104,
	"capture_time_utc": 1607360235.925104,
	"ecef": [
		3964520.64,
		378870.2,
		4965367.71
	],
	"gdop": 4.19,
	"llh": [
		51.45557,
		5.4589,
		107.4
	],
	"timestamp": 1607360244
}
```

vs `wifi`:

```json
{
	"accuracy": 19,
	"algorithm_type": "wifi",
	"llh": [
		51.455751,
		5.459112,
		0
	],
	"timestamp": 1607359974
}
```